presto-main/src/main/java/com/facebook/presto/operator/scalar/TDigestFunctions.java

```diff
@@ -181,4 +182,32 @@ public static Slice constructTDigest(
 
         return tDigest.serialize();
     }
+
+    @ScalarFunction(value = "array_merge", visibility = EXPERIMENTAL)
+    @Description("Merge an array of TDigests into a single TDigest")
+    @SqlType("tdigest(double)")
+    @SqlNullable
+    public static Slice arrayMerge(@SqlType("array(tdigest(double))") Block input)
+    {
+        if (input.getPositionCount() == 0) {
+            return null;
+        }
+
+        int firstNonNullIndex = 0;
+        while (firstNonNullIndex < input.getPositionCount() && input.isNull(firstNonNullIndex)) {
+            firstNonNullIndex++;
+        }
+        if (firstNonNullIndex == input.getPositionCount()) {
+            return null;
+        }
+        Slice initialSlice = input.getSlice(firstNonNullIndex, 0, input.getSliceLength(firstNonNullIndex));
```

Comment:
@tdcmeehan - responding to the comment you originally left in https://github.com/prestodb/presto/pull/20510: 

> Why treat the first null separately from the rest?

No major reason aside from that I felt that it was safer to look for the first non-null slice at the expense of readability. If you feel strongly against this, I can write it more like this:
```java
if (input.getPositionCount() == 0) {
    return null;
}

TDigest output = null;
for (int i = 0; i < input.getPositionCount(); i++) {
    if (input.isNull(i)) {
        continue;
    }
    if (output != null) {
        output.merge(createTDigest(input.getSlice(i, 0, input.getSliceLength(i))));
    }
}

return output.serialize();
```